packages/n4s/src/enforce/compounds/oneOf.js

```diff
@@ -0,0 +1,9 @@
+import runLazyRules from 'runLazyRules';
+
+/**
+ * @param {*} value   Value to be test against rules
+ * @param {Function[]} rules    Rules to validate the value with
+ */
+export default function oneOf(value, ...rules) {
+  return (rules) && (rules.filter(ruleGroup => runLazyRules(ruleGroup, value)).length === 1);
```

Comment:
Cleaver use of filter! Nice!
It could be a bit wasteful and keep running even after more than two tests passed. I would have preferred to short circuit immediately as we detect more than one successful test.
Maybe we could try something similar to [this](https://github.com/ealush/vest/blob/latest/packages/anyone/src/one.js)?